### ISSUE TITLE
Add Node2D.get_global_rot method ?

### ISSUE THREAD

**Issue by touilleMan** [godotengine/godot#6148]
Hi,

`Node2D` has `get_pos` and `get_global_pos` to get position relative to the parent node or global to the scene.
On the other hand it seems it only have `get_rot`, so why not add a `get_global_rot` ?

Beside it implementation would be pretty similar to `get_global_pos`:

``` cpp
float Node2D::get_global_rot() const {

    return get_global_transform().get_rotation();
}
```

I can submit a PR if you're interested ;-)

**Comment by bojidar-bg** [comment]
Note that if such a method is added, the natural question that follows is "Would we have get_global_scale in Node2D?"

I personally don't really see the need of it, i.e. when would one want only the rotation, and not scale and so on. Also, for some transformations (like top-down -> isometric), the rotation information is a bit speculative.

**Comment by touilleMan** [comment]
My need was pretty simple: I wanted to send a projectile from a moving/rotating character.
To do so I get the position and rotation of the firing character and apply them to a new instance of the projectile:

```gdscript

extends "base_weapon.gd"

export(int) var speed
export(PackedScene) var projectile_scene

func _fire():
	var projectile = projectile_scene.instance()
	projectile.set_pos(self.get_global_pos())
	var orientation = get_global_transform().get_rotation()
	projectile.set_rot(orientation)
	var linear_speed = Vector2(cos(orientation) * speed, -sin(orientation) * speed)
	projectile.set_linear_velocity(linear_speed)
	get_tree().get_root().add_child(projectile)
```

Given I could really easily get the global position with `get_global_pos`, it seems pretty legit to me you can get the same for the rotation...
But maybe I'm missing the obvious way to do that in Godot :wink:

**Comment by EdwardAngeles** [comment]
Since there is already a `get_global_pos()` function, the way I see it `get_global_rotation()` should be added as well as `get_global_scale()`.

**Comment by bojidar-bg** [comment]
@EdwardAngeles Slight nitpick, `get_global_rot` is better.

Also, we would have setters of the same properties.

@touilleMan Sounds legit, though you can just make `projectile.set_global_transform(get_global_transform())` and be done with it.

Sounds good to me though, better to have those 4 methods than not, given that they are pretty easy to implement and would lead to more API coherence.

**Comment by touilleMan** [comment]
I've added a PR on this :+1:

**Comment by rgrams** [comment]
Well this would revolutionize my coding experience! I can't count how many hours I've wasted fighting with global vs. local transforms.

**Comment by touilleMan** [comment]
implemented by c402ac13df1a945336faa009b469ac4f0528770f
